src/main/scala/tile/RocketTile.scala

```diff
@@ -83,6 +88,29 @@ class RocketTile private(
     beu
   }
 
+  /**
+   * A Centralized Trace encoder controller,
+   * controlling enable/disable of trace encoder
+   * and selecting trace sink
+   */
+  val trace_encoder_controller = rocketParams.ltrace.map { t =>
+    val trace_encoder_controller = LazyModule(new TraceEncoderController(t.encoderBaseAddr, xBytes))
+    connectTLSlave(trace_encoder_controller.node, xBytes)
+    trace_encoder_controller
+  }
+
+  val (trace_sinks, trace_sink_ids) = rocketParams.ltrace match {
+    case Some(t) => 
```

Comment:
I
```suggestion
    case Some(t) => t.buildSinks.map {(p)}.unzip
```

Does this work?